### PR TITLE
Added a 'testingEnrollIntoABExperiment' and 'testingExitABExperiment' methods

### PR DESCRIPTION
**Pull request by ijunaid**

### CODE REVIEW

**Review by turtledreams — CHANGES_REQUESTED** [review]
this is a response I get for experiment information:
image

Here the key is Nightfox:
image

[experimentInformation.experimentName] is the experiment name. Currently exitABTestsForKeys works for experiment keys only not for experiment names (let me know if this changed). So I would assume this call would fail. What do you guys think? @ijunaid @peterBrxwn @ArtursKadikis

**Review by turtledreams — APPROVED** [review]

File: example/lib/page_remote_config.dart [hunk 1]
```diff
@@ -14,9 +14,9 @@ class RemoteConfigPage extends StatefulWidget {
 }
 
 class _RemoteConfigPageState extends State<RemoteConfigPage> {
+  String rcKey = 'testKey';
```
Comment (peterBrxwn) [inline example/lib/page_remote_config.dart:17]: does this value change? if it doesn't, use const

File: example/lib/page_remote_config.dart [hunk 2]
```diff
@@ -223,33 +254,74 @@ class _RemoteConfigPageState extends State<RemoteConfigPage> {
...
+      result.forEach((key, value) {
+        resultString += '\n[$key]:\n';
+        value.forEach((item) {
+          resultString += '- [$item]\n';
+        });
+      });
+      if (context.mounted) {
+        showCountlyToast(context, resultString, null);
+      }
+      print(resultString);
```
Comment (peterBrxwn) [inline example/lib/page_remote_config.dart:269]: For consistency with line 280, this print should be before line 266

File: example/lib/page_remote_config.dart [hunk 3]
```diff
@@ -223,33 +254,74 @@ class _RemoteConfigPageState extends State<RemoteConfigPage> {
...
-        if (rResult == RequestResult.success) {
-          Map<String, ExperimentInformation> experimentInfoMap = await Countly.instance.remoteConfig.testingGetAllExperimentInfo();
-          print(experimentInfoMap);
-          if (context.mounted) {
-            showCountlyToast(context, 'Experiment Info:${experimentInfoMap}', null);
-          }
+      String message = 'Downloaded experiment information';
+      Color? color = null;
+      Countly.instance.remoteConfig.testingDownloadExperimentInformation((rResult, error) {
+        if (error != null) {
```
Comment (peterBrxwn) [inline example/lib/page_remote_config.dart:293]: You should print `rResult` if error == null.

File: example/lib/page_remote_config.dart [hunk 4]
```diff
@@ -223,33 +254,74 @@ class _RemoteConfigPageState extends State<RemoteConfigPage> {
...
+          print('RCDownloadCallback, Result:[$rResult], error:[$error]');
+          message = 'Downloaded experiment information failed';
+          color = Colors.red;
         }
+        showCountlyToast(context, message, color);
```
Comment (peterBrxwn) [inline example/lib/page_remote_config.dart:298]: I think we should check `if (context.mounted)`

File: lib/remote_config_internal.dart [hunk 5]
```diff
@@ -361,16 +405,15 @@ class RemoteConfigInternal implements RemoteConfig {
...
+  Future<Map<String, ExperimentInformation>> testingGetAllExperimentInfo() async {
     if (!_countlyState.isInitialized) {
       Countly.log('"initWithConfig" must be called before "testingGetAllExperimentInfo"', logLevel: LogLevel.ERROR);
       return {};
     }
 
     final List<dynamic> experimentsInfo = await _countlyState.channel.invokeMethod('testingGetAllExperimentInfo');
     List<ExperimentInformation> experimentsInfoList = experimentsInfo.map((e) => ExperimentInformation.fromJson(e)).toList();
-    Map<String, ExperimentInformation> experimentsInfoMap = { for (var e in experimentsInfoList) e.experimentID : e };
+    Map<String, ExperimentInformation> experimentsInfoMap = {for (var e in experimentsInfoList) e.experimentID: e};
```
Comment (peterBrxwn) [inline lib/remote_config_internal.dart:416]: e should be `final` not `var`

**Review by peterBrxwn — APPROVED** [review]

File: example/lib/page_remote_config.dart (hunk 1 above)
Comment (turtledreams) [reply example/lib/page_remote_config.dart:17]: it does

File: example/lib/page_remote_config.dart (hunk 4 above)
Comment (turtledreams) [reply example/lib/page_remote_config.dart:298]: not in syn code

File: example/lib/page_remote_config.dart (hunk 3 above)
Comment (turtledreams) [reply example/lib/page_remote_config.dart:293]: im 50/50 on this

**Pull request merged**